guides/source/upgrading_ruby_on_rails.md

```diff
@@ -16,6 +16,21 @@ Before attempting to upgrade an existing application, you should be sure you hav
 
 The best way to be sure that your application still works after upgrading is to have good test coverage before you start the process. If you don't have automated tests that exercise the bulk of your application, you'll need to spend time manually exercising all the parts that have changed. In the case of a Rails upgrade, that will mean every single piece of functionality in the application. Do yourself a favor and make sure your test coverage is good _before_ you start an upgrade.
 
+### The Upgrade Process
+
+When changing Rails versions, it's best to move slowly, one minor version at a time, in order to make good use of the deprecation warnings. Rails version numbers are in the form Major.Minor.Patch. Major and Minor versions change the API which means you will probably need to change your code. Patch versions are just bug fixes that don't change the API.
```

Comment:
Major and Minor versions change the API which means you will probably need to change your code.
`Major and Minor versions are allowed to make changes to the public API, so this may cause errors in your application.`

Patch versions are just bug fixes that don't change the API.
`Patch versions only include bug fixes, and don't change any public API.`